Examples/Algorithms/TrackFindingExaTrkX/include/ActsExamples/TrackFindingExaTrkX/TrackFindingAlgorithmExaTrkX.hpp

```diff
@@ -83,13 +86,26 @@ class TrackFindingAlgorithmExaTrkX final : public IAlgorithm {
   ActsExamples::ProcessCode execute(
       const ActsExamples::AlgorithmContext& ctx) const final;
 
+  /// Finalize and print timing
+  ActsExamples::ProcessCode finalize() final;
+
   const Config& config() const { return m_cfg; }
 
  private:
-  // configuration
   Config m_cfg;
 
   Acts::ExaTrkXPipeline m_pipeline;
+  mutable std::mutex m_mutex;
+
+  using Accumulator = boost::accumulators::accumulator_set<
+      float, boost::accumulators::features<boost::accumulators::tag::mean,
+                                           boost::accumulators::tag::variance>>;
+
+  mutable struct {
```

Comment:
`mutable` because we need to modify this in `execute` right?

Reply:
yes, otherwise we cannot aggregate this... This is a common pattern in the examples framework I think...